### ISSUE TITLE
Reword "Spam is not permitted here"

### ISSUE THREAD

**Issue by Yoric** [matrix-org/synapse#9830]
This phrasing is a bit too categorical, especially if for some reason we end up accidentally blocking something that is not actual spam.

Cc @deniseal to discuss a better phrasing.

**Comment by anoadragon453** [comment]
This error is returned to the user when the `check_event_for_spam` method reports that an event is not allowed.

Note that if you return a str from `check_event_for_spam`, that str will be returned to the user instead of the generic "Spam is not permitted here" error message.

If this is more about rebranding the spam checker as a whole, I'd probably meld this into #9805.

**Comment by Yoric** [comment]
> This error is returned to the user when the `check_event_for_spam` method reports that an event is not allowed.

Indeed, but that remains a very harsh message. I don't want to be insulted by my software, especially since it may commit errors, once in a while.

> Note that if you return a `str` type from `check_event_for_spam` instead of a `bool`, that `str` will be returned to the user instead of the generic "Spam is not permitted here" error message.

Indeed, this means that we can kinda walk around the issue. Still not ideal :)

As a side-note, none of this supports internationalization, which is another problem.

**Comment by anoadragon453** [comment]
The typical fix for internationalisation (and perhaps the original problem) is to return a specific errcode (`M_SPAM` or similar) such that the client could then generate an appropriate error message once it realised why the event was rejected.

We don't really have any errcodes that seem suitable at the moment however: https://spec.matrix.org/unstable/client-server-api/#api-standards

Anyways, that may be the long-term goal, but for now we can certainly change the default error message.

**Comment by Yoric** [comment]
Won't solve the problem in the short run, but I've just opened [an MSC for a rejection errcode](https://github.com/matrix-org/matrix-doc/pull/3174).
